core/extension/gdnative_interface.cpp

```diff
@@ -0,0 +1,685 @@
+	String *dest = (String *)r_dest;
+	memnew_placement(dest, String);
+	dest->parse_utf16(p_contents, p_size);
+}
+
+static void gdnative_string_new_with_utf32_chars_and_len(GDNativeStringPtr r_dest, const char32_t *p_contents, const GDNativeInt p_size) {
+	String *dest = (String *)r_dest;
+	memnew_placement(dest, String);
+	*dest = String((const char32_t *)p_contents, p_size);
+}
+
+static void gdnative_string_new_with_wide_chars_and_len(GDNativeStringPtr r_dest, const wchar_t *p_contents, const GDNativeInt p_size) {
+	String *dest = (String *)r_dest;
+	if (sizeof(wchar_t) == 2) {
+		// wchar_t is 16 bit, parse.
+		memnew_placement(dest, String);
+		dest->parse_utf16((const char16_t *)p_contents, p_size);
+	} else {
+		// wchar_t is 32 bit, copy.
+		memnew_placement(dest, String);
+		*dest = String((const char32_t *)p_contents, p_size);
+	}
+}
+
+static GDNativeInt gdnative_string_to_latin1_chars(const GDNativeStringPtr p_self, char *r_text, GDNativeInt p_max_write_length) {
+	String *self = (String *)p_self;
+	CharString cs = self->ascii(true);
+	GDNativeInt len = cs.length();
+	if (r_text) {
+		const char *s_text = cs.ptr();
+		for (GDNativeInt i = 0; i < MIN(len, p_max_write_length); i++) {
+			r_text[i] = s_text[i];
+		}
+	}
+	return len;
+}
+static GDNativeInt gdnative_string_to_utf8_chars(const GDNativeStringPtr p_self, char *r_text, GDNativeInt p_max_write_length) {
+	String *self = (String *)p_self;
+	CharString cs = self->utf8();
+	GDNativeInt len = cs.length();
+	if (r_text) {
+		const char *s_text = cs.ptr();
+		for (GDNativeInt i = 0; i < MIN(len, p_max_write_length); i++) {
+			r_text[i] = s_text[i];
+		}
+	}
+	return len;
+}
+static GDNativeInt gdnative_string_to_utf16_chars(const GDNativeStringPtr p_self, char16_t *r_text, GDNativeInt p_max_write_length) {
+	String *self = (String *)p_self;
+	Char16String cs = self->utf16();
+	GDNativeInt len = cs.length();
+	if (r_text) {
+		const char16_t *s_text = cs.ptr();
+		for (GDNativeInt i = 0; i < MIN(len, p_max_write_length); i++) {
+			r_text[i] = s_text[i];
+		}
+	}
+	return len;
+}
+static GDNativeInt gdnative_string_to_utf32_chars(const GDNativeStringPtr p_self, char32_t *r_text, GDNativeInt p_max_write_length) {
+	String *self = (String *)p_self;
+	GDNativeInt len = self->length();
+	if (r_text) {
+		const char32_t *s_text = self->ptr();
+		for (GDNativeInt i = 0; i < MIN(len, p_max_write_length); i++) {
+			r_text[i] = s_text[i];
+		}
+	}
+	return len;
+}
+static GDNativeInt gdnative_string_to_wide_chars(const GDNativeStringPtr p_self, wchar_t *r_text, GDNativeInt p_max_write_length) {
+	if (sizeof(wchar_t) == 4) {
+		return gdnative_string_to_utf32_chars(p_self, (char32_t *)r_text, p_max_write_length);
+	} else {
+		return gdnative_string_to_utf16_chars(p_self, (char16_t *)r_text, p_max_write_length);
+	}
+}
+
+static char32_t *gdnative_string_operator_index(GDNativeStringPtr p_self, GDNativeInt p_index) {
+	String *self = (String *)p_self;
+	return self->ptrw();
```

Comment:
p_index is unused. Same with cost operator. 

```suggestion
	return self->ptrw()[p_index];
```

Reply:
Thanks! All these functions were already like this in GDNative and it seems they are all broken, I copied them without modification. At least they will work ok now :)